eng/pipelines/libraries/build-job.yml

```diff
@@ -40,8 +47,10 @@ jobs:
           - ${{ if ne(parameters.framework, 'allConfigurations') }}:
             - _skipTestRestoreArg: /p:SkipTestRestore=false
 
+        - ${{ parameters.variables }}
+
       steps:
-        - script: $(_buildScript) -restore $(_buildArguments) $(_skipTestRestoreArg)
+        - script: $(_buildScript) -restore $(_buildArguments) $(_skipTestRestoreArg) $(_coreClrOverridePathArg)
```

Comment:
I suggest the _coreClrOverridePathArg to be included in the `_buildArguments` directly as it needs to be passed down for the build step as well. The reason for that is the way currently libraries resolves coreclr. During the build, if we find a `ReferenceFromRuntime=System.Private.CoreLib` we will transform that to a `ProjectReference=runtime.depproj` which will restore coreclr's transport package by default if no CoreCLROVerridePath is passed down. So we need to pass it down to the build for the projects that reference corelib from runtime to restore the correct S.P.CoreLib.

This will change with @jkoritzinsky change as the goal is that coreclr just binplaces the assets directly where libraries expects them or to create a convention in between the two of them so that we no longer need CoreCLROverridePath.